lib/discover/local.go

```diff
@@ -197,6 +197,21 @@ func (c *localClient) registerDevice(src net.Addr, device Device) bool {
 		}
 
 		if len(tcpAddr.IP) == 0 || tcpAddr.IP.IsUnspecified() {
+			srcAddr, err := net.ResolveTCPAddr("tcp", src.String())
```

Comment:
Maybe. How do I get the IP address from a net.Addr? For net.TCPAddr it's simple.